src/daft-logical-plan/src/ops/pivot.rs

```diff
@@ -32,14 +33,30 @@ impl Pivot {
         names: Vec<String>,
     ) -> logical_plan::Result<Self> {
         let upstream_schema = input.schema();
-        let (group_by, group_by_fields) =
-            resolve_exprs(group_by, &upstream_schema, false).context(CreationSnafu)?;
-        let (pivot_column, _) =
-            resolve_single_expr(pivot_column, &upstream_schema, false).context(CreationSnafu)?;
-        let (value_column, value_col_field) =
-            resolve_single_expr(value_column, &upstream_schema, false).context(CreationSnafu)?;
-        let (aggregation, _) =
-            resolve_single_aggexpr(aggregation, &upstream_schema).context(CreationSnafu)?;
+
+        let expr_resolver = ExprResolver::default();
+        let agg_resolver = ExprResolver::builder().in_agg_context(true).build();
+
+        let (group_by, group_by_fields) = expr_resolver
+            .resolve(group_by, &upstream_schema)
+            .context(CreationSnafu)?;
+        let (pivot_column, _) = expr_resolver
+            .resolve_single(pivot_column, &upstream_schema)
+            .context(CreationSnafu)?;
+        let (value_column, value_col_field) = expr_resolver
+            .resolve_single(value_column, &upstream_schema)
+            .context(CreationSnafu)?;
+
+        let (aggregation, _) = agg_resolver
+            .resolve_single(aggregation, &upstream_schema)
+            .context(CreationSnafu)?;
+
+        let Expr::Agg(agg_expr) = aggregation.as_ref() else {
```

Comment:
use `Arc::unwrap_or_clone`
```suggestion
        let Expr::Agg(agg_expr) = Arc::unwrap_or_clone(aggregation) {
```

Reply:
I added a more informative error message that uses `aggregation` now, so an `unwrap_or_clone`, which consumes `aggregation`, cannot be used anymore.